packages/core/src/modules/connections/services/ConnectionService.ts

```diff
@@ -403,6 +407,31 @@ export class ConnectionService {
     return connection
   }
 
+  /**
+   * Process a received {@link ProblemReportMessage}.
+   *
+   * @param messageContext The message context containing a connection problem report message
+   * @returns connection record associated with the connection problem report message
+   *
+   */
+  public async processProblemReport(
+    messageContext: InboundMessageContext<ConnectionProblemReportMessage>
+  ): Promise<ConnectionRecord> {
+    const { message: connectionProblemReportMessage, connection, recipientVerkey } = messageContext
+
+    this.logger.debug(`Processing connection problem report for verkey ${recipientVerkey}`)
+
+    if (!connection) {
```

Comment:
There is a good chance the `connection` won't be availalbe on the `messageContext` yet as the connection is not complete yet. If you look at the other methods in the conenction service you can see they often retrieve it from the repository. This is only the case in the connection implementation as before the protocol is finished there is no complete connection yet.